openrtb/category.go

```diff
@@ -422,3 +424,432 @@ const (
+	string(CategoryJapan):                   CategoryJapan,
+	string(CategoryMexicoAndCentralAmerica): CategoryMexicoAndCentralAmerica,
+	string(CategoryNationalParks):           CategoryNationalParks,
+	string(CategorySouthAmerica):            CategorySouthAmerica,
+	string(CategorySpas):                    CategorySpas,
+	string(CategoryThemeParks):              CategoryThemeParks,
+	string(CategoryTravelingwithKids):       CategoryTravelingwithKids,
+	string(CategoryUnitedKingdom):           CategoryUnitedKingdom,
+
+	string(CategoryRealEstate):         CategoryRealEstate,
+	string(CategoryApartments):         CategoryApartments,
+	string(CategoryArchitects):         CategoryArchitects,
+	string(CategoryBuyingSellingHomes): CategoryBuyingSellingHomes,
+
+	string(CategoryShopping):            CategoryShopping,
+	string(CategoryContestsAndFreebies): CategoryContestsAndFreebies,
+	string(CategoryCouponing):           CategoryCouponing,
+	string(CategoryComparison):          CategoryComparison,
+	string(CategoryEngines):             CategoryEngines,
+
+	string(CategoryReligionAndSpirituality): CategoryReligionAndSpirituality,
+	string(CategoryAlternativeReligions):    CategoryAlternativeReligions,
+	string(CategoryAtheismAgnosticism):      CategoryAtheismAgnosticism,
+	string(CategoryBuddhism):                CategoryBuddhism,
+	string(CategoryCatholicism):             CategoryCatholicism,
+	string(CategoryChristianity):            CategoryChristianity,
+	string(CategoryHinduism):                CategoryHinduism,
+	string(CategoryIslam):                   CategoryIslam,
+	string(CategoryJudaism):                 CategoryJudaism,
+	string(CategoryLatterDaySaints):         CategoryLatterDaySaints,
+	string(CategoryPaganWiccan):             CategoryPaganWiccan,
+
+	string(CategoryUncategorized): CategoryUncategorized,
+
+	string(CategoryNonStandardContent):             CategoryNonStandardContent,
+	string(CategoryUnmoderatedUGC):                 CategoryUnmoderatedUGC,
+	string(CategoryExtremeGraphicExplicitViolence): CategoryExtremeGraphicExplicitViolence,
+	string(CategoryPornography):                    CategoryPornography,
+	string(CategoryProfaneContent):                 CategoryProfaneContent,
+	string(CategoryHateContent):                    CategoryHateContent,
+	string(CategoryUnderConstruction):              CategoryUnderConstruction,
+	string(CategoryIncentivized):                   CategoryIncentivized,
+
+	string(CategoryIllegalContent):        CategoryIllegalContent,
+	string(CategoryIllegalContent2):       CategoryIllegalContent2,
+	string(CategoryWarez):                 CategoryWarez,
+	string(CategorySpywareMalware):        CategorySpywareMalware,
+	string(CategoryCopyrightInfringement): CategoryCopyrightInfringement,
+}
+
+// MarshalCategory converts a string into an Category, and returns an error if
+// there is no corresponding Category.
+func MarshalCategory(value string) (Category, error) {
```

Comment:
marshal sounds like convert some type to string, but it's doing opposite here.

Reply:
Okay I will change it to UnmarshalToCategory